llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -17206,6 +17210,21 @@ SDValue DAGCombiner::visitFP_TO_UINT(SDNode *N) {
   return FoldIntToFPToInt(N, DAG);
 }
 
+SDValue DAGCombiner::visitXRINT(SDNode *N) {
+  SDValue N0 = N->getOperand(0);
+  EVT VT = N->getValueType(0);
+
+  // fold (lrint|llrint undef) -> undef
+  if (N0.isUndef())
+    return DAG.getUNDEF(VT);
+
+  // fold (lrint|llrint c1fp) -> c1
+  if (DAG.isConstantFPBuildVectorOrConstantFP(N0))
+    return DAG.getNode(N->getOpcode(), SDLoc(N), VT, N0);
```

Comment:
Do we have lrint/llrint constant folding?

Reply:
No. Do you suggest adding it to the list in `canConstantFoldCallTo`?

Reply:
I guest RKSimon means is there folding similar with FoldIntToFPToInt?

Reply:
It is impossible to recover the FP after converting FP to int with rounding, so I don't think we can do any constant folding similar to `FoldIntToFPToInt`. I've investigated `canConstantFoldCallTo`, and it doesn't seem to be of particular interest in the context of [l]lrint either.